### ISSUE TITLE
The loss function of inner loop

### ISSUE THREAD

**Issue by pipiPdesu** [ruoxi-jia-group/ASSET#2]
Thanks for sharing your amazing work!

I noticed that the loss function of the inner loop shoulde be binary cross entropy loss but in demo it was assigned as MSEloss, I wonder is it has a side effect on the final result.

![image](https://github.com/ruoxi-jia-group/ASSET/assets/54018541/5a902aa6-5a1e-4fca-8821-f80fb16e3352)

Additionally, I've noted that the 'o_model', which was referenced in #1 and utilized in the inner loop, is trained with a poison rate of 0.1, while the 'train_dataset' in the demo has a poison rate of 0.05. My understanding is that the 'o_model' should be trained on the same poisoned 'train_dataset'. However, it appears that they are not identical. I'm wondering if this discrepancy could also influence the final results.
